### DISCUSSION TITLE
Doris 创建MySql 外表同步，json格式字段转成字符格式，查询后字段内容乱码

### DISCUSSION THREAD

**Discussion by libra612** [apache/doris#9543] [A - General / Q&A]
mysql库字段格式设置为json，doris外表字段格式设置为varchar,查询doris外表，该字段内容显示为乱码

**Comment by hf200012** [comment]
doris是utf8编码，你MySQL字符集是什么，或者你入doris的时候做转码了没

**Reply by libra612** [reply]
两边都是utf-8字符集，mysql表字段格式为json，存储内容为{"lat": 30.607453173417365, "lng": 114.27746609816468, "city": "420100", "guid": "4a2be456058744acb04b2736afa3cef3", "space": 1.9642, "title": "XX医院", "district": "", "province": "420000", "addressAll": "湖北省武汉市江汉区新华路255号", "mapPointName": "湖北省武汉市江汉区解放大道1277号"}，doris外表显示内容为7B226C6174223A2033302E3630373435333137333431373336352C20226C6E67223A203131342E3237373

**Reply by hf200012** [reply]
你的同步方式是什么

**Comment by libra612** [comment]
在doris里面创建mysql的外表，mysql版本是8.0，字段格式设置为json格式，doris外表格式设置为字符，查询doris的mysql外表，json格式字段显示的内容不再是原始的json数据
















在 2022-05-31 09:16:14，"jiafeng.zhang" ***@***.***> 写道：

你的同步方式是什么
